specs/progress.md

```diff
@@ -0,0 +1,142 @@
+# fast-progress
+
+## Overview
+
+*Progress* is used to display the length of a process or to represent an unspecified wait time. *Progress* can either be circular or linear. *Progress* can either be in an **indeterminate** or **determinate** state. This is determined by whether `value` attribute is passed to the component. 
+
+### Use Cases
+
+The *progress* component should be used for an element that displays the progress status for a task that takes a long time or consists of several steps. Typical use cases include loading elements or status of a download. 
+  
+### Features
+
+A progress should allow the following properties:
+- `value`, the value of the progress, if not passed the progress will be put into its "indeterminate" state.
+- `min-value`, the minimum value.
+- `max-value`, the maximum value.
+- `circular`, whether the progress renders circular or not.
+- `paused`, whether the progress is paused or not.
```

Comment:
What's the use case for paused? In the determinate state the value would just not be changing, in the indeterminate state I can only assume the animation would freeze which would be odd behavior.